features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java

```diff
@@ -0,0 +1,166 @@
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.events.commands;
+
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Argument;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.apache.karaf.shell.api.console.Terminal;
+import org.apache.karaf.shell.support.table.ShellTable;
+import org.opennms.core.criteria.Alias;
+import org.opennms.core.criteria.CriteriaBuilder;
+import org.opennms.netmgt.dao.api.EventDao;
+import org.opennms.netmgt.model.OnmsAlarm;
+import org.opennms.netmgt.model.OnmsDistPoller;
+import org.opennms.netmgt.model.OnmsEvent;
+import org.opennms.netmgt.model.OnmsEventParameter;
+import org.opennms.netmgt.model.OnmsNode;
+import org.opennms.netmgt.model.OnmsServiceType;
+import org.opennms.netmgt.model.OnmsSeverity;
+
+@Command(scope = "opennms", name = "event", description = "Show event details for a given id")
+@Service
+public class EventCommand implements Action {
+
+    @Reference
+    public EventDao eventDao;
+
+    @Argument(name = "id", description = "Event Id to match (exact).", required = false, multiValued = false)
+    Integer id;
+
+    @Reference(optional = true)
+    Terminal terminal;
+
+    @Override
+    public Object execute() {
+        final CriteriaBuilder criteriaBuilder = new CriteriaBuilder(OnmsEvent.class)
+                .orderBy("eventTime").desc()
+                .eq("id", id)
+                .alias("node", "node", Alias.JoinType.LEFT_JOIN)
+                .alias("alarm", "alarm", Alias.JoinType.LEFT_JOIN)
+                .alias("eventParameters", "event_parameters", Alias.JoinType.LEFT_JOIN)
+                .alias("serviceType", "serviceType", Alias.JoinType.LEFT_JOIN);
+
+        final var onmsEvents = eventDao.findMatching(criteriaBuilder.toCriteria());
+
+        if (onmsEvents.size() == 0) {
+            return null;
+        }
+
+        final OnmsEvent onmsEvent = onmsEvents.get(0);
+
+        final ShellTable eventPropertyTable = new ShellTable();
+        eventPropertyTable.size(terminal.getWidth() - 1);
```

Comment:
If `terminal` is `optional`, could it be null here?

Reply:
done